### ISSUE TITLE
[Julia] Implement new Concept Exercise: primitive-types

### ISSUE THREAD

**Issue by SaschaMann** [exercism/v3#831]
This issue describes how to implement the `primitive-types` concept exercise for the Julia track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

## Goal

The goal of this exercise is to teach the student how primitive types are implemented in Julia. We will teach primitive types through Julia's `primitive` type.

## Learning objectives

- Know of the existence of user defined primitive types.
- Know how to define a primitive type.

## Concepts

- `primitive-types`

## Prerequisites

This will likely be one of the last exercises of the `type-system` block, so you can assume knowledge of most `type-system` concepts.

- `methods`
- `bitwise-operation`
- `string-literals` (optional, see below)
- `abstract-types`
- `composite-types`
- `mutable-composite-types`

## Resources to refer to

- [Julia Manual](https://docs.julialang.org/en/v1/manual/types/#Primitive-Types-1)
- [BioSymbols.jl](https://github.com/BioJulia/BioSymbols.jl)

### Hints

- [Julia Manual](https://docs.julialang.org/en/v1/manual/types/#Primitive-Types-1)

## Implementing

- [How to implement a Concept Exercise?](https://exercism.github.io/v3/#/languages/julia/docs/implementing-a-concept-exercise)

The context should be DNA/RNA types.

You can assume that the student has solved the v2 biology-themed exercises [`rna-transcription`]() ([extracted concepts](https://github.com/exercism/v3/blob/master/languages/julia/reference/exercise-concepts/rna-transcription.md)), [`hamming`](https://github.com/exercism/julia/tree/master/exercises/hamming) and [`nucleotide-count`](https://github.com/exercism/julia/tree/master/exercises/nucleotide-count) using strings and characters.

This exercise should have the student define their own DNA/RNA types, based on `BioSymbol.jl`'s definitions, and then be able to solve the exercises above using the newly defined types rather than strings/chars. For convenience, they should also define a method to convert strings/chars to the new types, this *could* also be a string literal, though that may make the exercise needlessly complicated (hard to judge ahead of time).

This exercise should only require solving **one** of the v2 exercises while the other two will be unlocked as practice exercises once this is completed. It's probably best to implement testsuites and examples for all three exercises before writing the instructions, so that one can pick which one of them is most suitable as concept exercise for primitive types.

**Note: I'm not a biologist, so perhaps this narrative idea is nonsense. Please correct me if that's the case!**

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by SaschaMann** [comment]
Apparently this is a feature one should never really use, outside of bootstrapping Julia. Most people on the Julia Slack strongly recommended against teaching this. We might pick this up later as an optional exercise with a huge warning sign, but for now primitive-types are out of scope.
